### ISSUE TITLE
Authoritative slave w/ presigned shows only one algorthm DS records in `pdnsutil show-zone`

### ISSUE THREAD

**Issue by gertvdijk** [PowerDNS/pdns#7667]
- Program: Authoritative 
 - Issue type: Bug report 

### Short description

PowerDNS master with DNSSEC key material and PowerDNS slave receiving the zones presigned show inconsistent results for `pdnsutil show-zones` on DS RRSets if more than one CSK is active.

This may result in providing incomplete DNSSEC details to a parent zone when the following conditions are met: 1) running the `pdnsutil show-zones` command from a slave and 2) if more than one CSK is active and 3) if the parent zone takes DS records as direct input (some/mine take DNSKEY instead, not sure what's the industry standard?).

### Environment

 - Operating system: Debian Stretch
 - Software version: 4.2.0-rc1 (4.1.8 probably also affected)
 - Software source: PowerDNS repository 

### Steps to reproduce

1. Set up a master with a zone.
2. Secure the zone with the defaults `pdns secure-zone ...`.
3. Add another CSK as active key, e.g. `pdnsutil add-zone-key mydomain.tld ksk active ed25519` and rectify the zone, etc.
4. Let the slave replicate over AXFR, receiving the zone presigned.
5. Run `pdnsutil show-zone mydomain.tld` on the master. This shows the expected results, with both DNSKEYs and two sets of DS RRsets.
   ```
   [...]
   Zone has hashed NSEC3 semantics, configuration: 1 0 100 cc816e8b4ae518d0c3f3b30ec4984616
   keys: 
   ID = 3 (CSK), flags = 257, tag = 65310, algo = 13, bits = 256     Active ( ECDSAP256SHA256 ) 
   CSK DNSKEY = mydomain.tld. IN DNSKEY 257 3 13 BDBMXa2+W4o6Pg5IqwbJSnOqRAnGNdV0fAPgHXa+/Eip5fHPqi0MPsm8V229OoC339la+frKhynxxN6G3+qjnw== ; ( ECDSAP256SHA256 )
   DS = mydomain.tld. IN DS 65310 13 1 a72ae702d04f351c9e5f202cfbaae9e407bfa118 ; ( SHA1 digest )
   DS = mydomain.tld. IN DS 65310 13 2 094c08929d854acd0197b38bb94d06453a4bc7601cb2fd40330e77fd852396a3 ; ( SHA256 digest )
   DS = mydomain.tld. IN DS 65310 13 4 9cc7098715dbe42d48fe0794baa198447dfbe365387e1f27af3d35a5fe7dedac4771baa7d037729f176b24f30231fdf9 ; ( SHA-384 digest )
   ID = 4 (CSK), flags = 257, tag = 58127, algo = 15, bits = 256     Active ( ED25519 ) 
   CSK DNSKEY = mydomain.tld. IN DNSKEY 257 3 15 bQHhJb4RJteFOagLrX+LEEprJ73urBG0q5Tu/2wizNk= ; ( ED25519 )
   DS = mydomain.tld. IN DS 58127 15 1 9d0b704d759b6e4c00226d3b7d91ef8cdd5f5620 ; ( SHA1 digest )
   DS = mydomain.tld. IN DS 58127 15 2 894ac4b4d20793aec29c4815b0165c1c55aae59a07b4c44c132aa3dcd4a55c93 ; ( SHA256 digest )
   DS = mydomain.tld. IN DS 58127 15 4 76292d6a869469278cc5e80c42b2b2d1a69c34b3ad3566efc691e9d33c71d19c341377066711fddc142f778d458a5a2d ; ( SHA-384 digest )
   ```
6. Run `pdnsutil show-zone mydomain.tld` on the slave. This shows both DNSKEYs but only one of the two DS RRsets (those for algorithm 13 are missing below):
   ```
   [...]
   Zone is presigned
   Zone has hashed NSEC3 semantics, configuration: 1 0 100 cc816e8b4ae518d0c3f3b30ec4984616
   keys: 
   KSK, tag = 58127, algo = 15, bits = 256
   DNSKEY = mydomain.tld. IN DNSKEY 257 3 15 bQHhJb4RJteFOagLrX+LEEprJ73urBG0q5Tu/2wizNk=; ( ED25519 ) 
   DS = mydomain.tld. IN DS 58127 15 1 9d0b704d759b6e4c00226d3b7d91ef8cdd5f5620 ; ( SHA1 digest )
   DS = mydomain.tld. IN DS 58127 15 2 894ac4b4d20793aec29c4815b0165c1c55aae59a07b4c44c132aa3dcd4a55c93 ; ( SHA256 digest )
   DS = mydomain.tld. IN DS 58127 15 4 76292d6a869469278cc5e80c42b2b2d1a69c34b3ad3566efc691e9d33c71d19c341377066711fddc142f778d458a5a2d ; ( SHA-384 digest )
   KSK, tag = 65310, algo = 13, bits = 256
   DNSKEY = mydomain.tld. IN DNSKEY 257 3 13 BDBMXa2+W4o6Pg5IqwbJSnOqRAnGNdV0fAPgHXa+/Eip5fHPqi0MPsm8V229OoC339la+frKhynxxN6G3+qjnw==; ( ECDSAP256SHA256 ) 
   ```
   <sup>The CSK shows up as KSK here, but I can understand that this does not matter to a slave, so it's not that important.</sup>

### Expected behaviour

The slave to synthesize all DS records for multiple DNSKEYs.

### Actual behaviour

*shown in steps to reproduce*

### Other information

* I have only tested CSK setup, split-key setups *may* also be affected.
* I have only tested 4.2.0-rc1 so far, 4.1.8 or older may also be affected.
* I have only tested with NSEC3 in regular non-narrow mode, but I think NSEC vs NSEC3 is not so relevant.

**Comment by Habbie** [comment]
reproduced on 490a22b04502f003770e555dddcefdfabda9de46 (4.2.0-rc one and a half)

**Comment by Habbie** [comment]
Fix in #7801
